app/src/main/java/com/x8bit/bitwarden/ui/platform/feature/vaultunlockednavbar/VaultUnlockedNavBarScreen.kt

```diff
@@ -424,9 +424,6 @@ private sealed class VaultUnlockedNavBarTab : Parcelable {
  */
 private fun NavController.vaultUnlockedNavBarScreenNavOptions(): NavOptions =
     navOptions {
-        popUpTo(graph.findStartDestination().id) {
-            saveState = true
-        }
+        popUpTo(graph.findStartDestination().id)
         launchSingleTop = true
-        restoreState = true
```

Comment:
So why do we need to update this stuff? My worry is that this won't maintain state across tabs, which I'm prettty sure is something we want.

Reply:
I took a quick stab at this and while I do find it odd that this problem exists at all, I did find a way around it without modifying the nav args. We can make the navigation state based:

```kotlin
data class VaultUnlockedNavBarState(
    @StringRes val vaultNavBarLabelRes: Int,
    @StringRes val vaultNavBarContentDescriptionRes: Int,
    val tabState: TabState,
) {
    sealed class TabState {
        data object Vault : TabState()

        data object Send : TabState()

        data object Generator : TabState()

        data object Settings : TabState()
    }
}
```

```kotlin
    LaunchedEffect(state.tabState) {
        navController.run {
            val navOptions = vaultUnlockedNavBarScreenNavOptions()
            when (state.tabState) {
                VaultUnlockedNavBarState.TabState.Vault -> navigateToVaultGraph(navOptions)
                VaultUnlockedNavBarState.TabState.Send -> navigateToSendGraph(navOptions)
                VaultUnlockedNavBarState.TabState.Generator -> navigateToGeneratorGraph(navOptions)
                VaultUnlockedNavBarState.TabState.Settings -> navigateToSettingsGraph(navOptions)
            }
        }
    }
```

This seemed to work really well.

Reply:
If we go a state-based route we have to be very careful and make sure there isn't a mismatch in the state when restoring after process death.

Reply:
> I do have another fix locked and loaded where the VM keeps hold of a SelectedBottomTab enum and no interaction with the navController takes place if its the currently selected one, this matches what is seen today in the settings screen where the user would just stay on the submenu view. 

Would adding the VM's state to SavedStateHandle take care of that @brian-livefront

Reply:
I don't think that is what he is talking about, I think he is concerned about the saved state in the `NavController`.

He wrote [the code](https://github.com/bitwarden/android/blob/2a057bb1fb2f368dd43042bc7f2179668893d9d6/app/src/main/java/com/x8bit/bitwarden/ui/platform/feature/rootnav/RootNavScreen.kt#L120) in the `RootNavScreen` to handle something similar.

Reply:
Brian did mention another way to we could manage this in the Screen, where we check to see if the newly selected destination is already the current graph and ignore the event in that case. That would require us to Expose route information though.

Reply:
Edit: I may have misunderstood him, I think he just wants to add this check in the `onClick` of the Navigation Item:


```kotlin
                onClick = {
                    if (isSelected) return@onClick <--- Here
                    when (destination) {
                        is VaultUnlockedNavBarTab.Vault -> vaultTabClickedAction()
                        VaultUnlockedNavBarTab.Send -> sendTabClickedAction()
                        VaultUnlockedNavBarTab.Generator -> generatorTabClickedAction()
                        VaultUnlockedNavBarTab.Settings -> settingsTabClickedAction()
                    }
                },
```

Reply:
I did some additional testing and found that the root cause of the problem seems to be that the Lifecycle state of the screen is `DESTROYED` when the tab gets clicked multiple times in a row. I don't entirely understand why that is, it may be a bug in Android Lifecycle. Adding the `isSelected` check may be the easiest way to work around it, though state based navigation would work as well but incurs a lot more effort.